### ISSUE TITLE
On providing options for rendering fenced code blocks

### ISSUE THREAD

**Issue by alakra** [pragdave/earmark#59]
In some code highlighting libraries, I have seen requirements expecting that the class name of a code block be prefixed with `language` like so:

```
<pre><code class="language-css">p { color: red }</code></pre>
```

In its current form, earmark doesn't appear to allow anything that doesn't match the regex pattern `\w*` when plucking the language type from the fence block token.  This eliminates hyphens from being matched so I can't do something like the following:

      ```language-ruby
      class Foo
        def bar
        end
      end
      ```

The `language-ruby` part will just get dropped completely by earmark.

After perusing the relevant modules, specifically `HtmlRenderer` and `Line`, I haven't seen a clear way to fix this without changing the code.  In my own fork, I changed the regex to accomplish what I needed for the short term, but it doesn't seem like the right way to fix this.  This seems like it should be fixed on the rendering portion of earmark.

Would it make sense to pass options to the html renderer to prefix code blocks with a static string or provide some callback function that change/transform it? Is there an existing way to accomplish this with the existing codebase?

Thanks for taking the time to look at this! :heart: :heart: :heart:

**Comment by RobertDober** [comment]
I guess it is useful and if it does not harm we should change the rgx. Maybe one day somebody will create a language named e.g. `better-language` and we would be ready :wink:  . Could you make a PR please, so that I can have a look. Thx.

**Comment by alakra** [comment]
@RobertDober Thanks for hearing me out. I'll put out a PR shortly.

**Comment by alakra** [comment]
See https://github.com/pragdave/earmark/pull/60

**Comment by RobertDober** [comment]
closed with PR #60
